vulkan.cppm compile errors

Hi,
I have compilation errors using the new vulkan.cppm on macOS with clang 16.0.6 from homebrew, CMake 3.26.4 and ninja 1.11.1. I am including the vulkan-headers repo with cmake's FetchContent.

Making the constants inline and removing the export keyword from `export namespace VULKAN_HPP_RAII_NAMESPACE` seems to fix it for me. Or is there a setting I am missing? 

Output:
```
viktormihalyi ~/vkcppmtest/build % cmake --build .
[4/8] Building CXX object CMakeFiles/VulkanHppModule.dir/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm.o
FAILED: CMakeFiles/VulkanHppModule.dir/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm.o CMakeFiles/VulkanHppModule.dir/vulkan.pcm 
/opt/homebrew/opt/llvm/bin/clang++  -I/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include -std=c++20 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.3.sdk -MD -MT CMakeFiles/VulkanHppModule.dir/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm.o -MF CMakeFiles/VulkanHppModule.dir/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm.o.d @CMakeFiles/VulkanHppModule.dir/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm.o.modmap -o CMakeFiles/VulkanHppModule.dir/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm.o -c /Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:2812:3: error: export declaration appears within another export declaration
  export namespace VULKAN_HPP_RAII_NAMESPACE
  ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:892:31: error: using declaration referring to 'ApiVersion' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::ApiVersion;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6941:29: note: target of using declaration
  VULKAN_HPP_CONSTEXPR auto ApiVersion            = makeApiVersion( 0, 1, 0, 0 );
                            ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:893:31: error: using declaration referring to 'ApiVersion10' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::ApiVersion10;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6942:29: note: target of using declaration
  VULKAN_HPP_CONSTEXPR auto ApiVersion10          = makeApiVersion( 0, 1, 0, 0 );
                            ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:894:31: error: using declaration referring to 'ApiVersion11' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::ApiVersion11;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6943:29: note: target of using declaration
  VULKAN_HPP_CONSTEXPR auto ApiVersion11          = makeApiVersion( 0, 1, 1, 0 );
                            ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:895:31: error: using declaration referring to 'ApiVersion12' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::ApiVersion12;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6944:29: note: target of using declaration
  VULKAN_HPP_CONSTEXPR auto ApiVersion12          = makeApiVersion( 0, 1, 2, 0 );
                            ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:896:31: error: using declaration referring to 'ApiVersion13' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::ApiVersion13;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6945:29: note: target of using declaration
  VULKAN_HPP_CONSTEXPR auto ApiVersion13          = makeApiVersion( 0, 1, 3, 0 );
                            ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:901:31: error: using declaration referring to 'AttachmentUnused' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::AttachmentUnused;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6866:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t AttachmentUnused                 = VK_ATTACHMENT_UNUSED;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:902:31: error: using declaration referring to 'False' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::False;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6867:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t False                            = VK_FALSE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:903:31: error: using declaration referring to 'HeaderVersion' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::HeaderVersion;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6891:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t HeaderVersion                    = VK_HEADER_VERSION;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:904:31: error: using declaration referring to 'HeaderVersionComplete' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::HeaderVersionComplete;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6946:29: note: target of using declaration
  VULKAN_HPP_CONSTEXPR auto HeaderVersionComplete = makeApiVersion( 0, 1, 3, VK_HEADER_VERSION );
                            ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:905:31: error: using declaration referring to 'LodClampNone' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::LodClampNone;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6868:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR float    LodClampNone                     = VK_LOD_CLAMP_NONE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:906:31: error: using declaration referring to 'LuidSize' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::LuidSize;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6869:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t LuidSize                         = VK_LUID_SIZE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:909:31: error: using declaration referring to 'MaxDescriptionSize' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::MaxDescriptionSize;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6870:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t MaxDescriptionSize               = VK_MAX_DESCRIPTION_SIZE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:910:31: error: using declaration referring to 'MaxDeviceGroupSize' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::MaxDeviceGroupSize;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6871:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t MaxDeviceGroupSize               = VK_MAX_DEVICE_GROUP_SIZE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:911:31: error: using declaration referring to 'MaxDriverInfoSize' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::MaxDriverInfoSize;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6872:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t MaxDriverInfoSize                = VK_MAX_DRIVER_INFO_SIZE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:912:31: error: using declaration referring to 'MaxDriverNameSize' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::MaxDriverNameSize;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6873:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t MaxDriverNameSize                = VK_MAX_DRIVER_NAME_SIZE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:913:31: error: using declaration referring to 'MaxExtensionNameSize' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::MaxExtensionNameSize;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6874:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t MaxExtensionNameSize             = VK_MAX_EXTENSION_NAME_SIZE;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:914:31: error: using declaration referring to 'MaxGlobalPrioritySizeKhr' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::MaxGlobalPrioritySizeKhr;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6875:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t MaxGlobalPrioritySizeKhr         = VK_MAX_GLOBAL_PRIORITY_SIZE_KHR;
                                ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.cppm:915:31: error: using declaration referring to 'MaxMemoryHeaps' with internal linkage cannot be exported
  using VULKAN_HPP_NAMESPACE::MaxMemoryHeaps;
                              ^
/Users/viktormihalyi/vkcppmtest/build/_deps/vulkan-headers-src/include/vulkan/vulkan.hpp:6876:33: note: target of using declaration
  VULKAN_HPP_CONSTEXPR uint32_t MaxMemoryHeaps                   = VK_MAX_MEMORY_HEAPS;
                                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
ninja: build stopped: subcommand failed.
```

This is the main.cpp:
```
#include <vector>

import vulkan;

int main ()
{
	vk::raii::Context context;
	vk::ApplicationInfo applicationInfo { "app", 1, "engine", 1, vk::ApiVersion10 };
	const std::vector<const char*> instanceExtensions { "VK_KHR_portability_enumeration" };
	vk::InstanceCreateInfo instanceCreateInfo { vk::InstanceCreateFlagBits::eEnumeratePortabilityKHR, &applicationInfo, {}, instanceExtensions };
	vk::raii::Instance instance { context, instanceCreateInfo };
	return 0;
}
```
And the CMakeLists.txt:
```
cmake_minimum_required (VERSION 3.25)
if (${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.26)
  set (CMAKE_EXPERIMENTAL_CXX_MODULE_CMAKE_API 2182bf5c-ef0d-489a-91da-49dbc3090d2a)
elseif (${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.25)
  set (CMAKE_EXPERIMENTAL_CXX_MODULE_CMAKE_API 3c375311-a3c9-4396-a187-3227ef642046)
endif ()

set (CMAKE_EXPERIMENTAL_CXX_MODULE_DYNDEP 1)
set (CMAKE_CXX_EXTENSIONS OFF)

include (FetchContent)

project (vkcppmtest)

set (CMAKE_CXX_STANDARD 20)
set (CMAKE_CXX_STANDARD_REQUIRED True)

FetchContent_Declare (
    Vulkan-Headers
    URL https://github.com/KhronosGroup/Vulkan-Headers/archive/refs/tags/v1.3.256.tar.gz
)
FetchContent_MakeAvailable (Vulkan-Headers)

add_library (VulkanHppModule)
target_include_directories (VulkanHppModule PRIVATE ${vulkan-headers_SOURCE_DIR}/include/)
target_sources (VulkanHppModule PRIVATE
    FILE_SET CXX_MODULES
    BASE_DIRS ${vulkan-headers_SOURCE_DIR}/include/
    FILES ${vulkan-headers_SOURCE_DIR}/include/vulkan/vulkan.cppm
)

add_executable (vkcppmtest)
target_sources (vkcppmtest PRIVATE src/main.cpp)
target_link_libraries (vkcppmtest PUBLIC VulkanHppModule)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -333,6 +333,20 @@ set( TINYXML2_SOURCES ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.cpp )
 set( TINYXML2_HEADERS ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.h )
 source_group( TinyXML2 FILES ${TINYXML2_HEADERS} ${TINYXML2_SOURCES} )
 
+# Build Vulkan-Hpp as a module
+if( VULKAN_HPP_ENABLE_EXPERIMENTAL_CPP20_MODULES )
+	# create a target to provide VulkanHpp as C++20 module
+	add_library( VulkanHppModule )
+	set_target_properties( VulkanHppModule PROPERTIES
+		CXX_STANDARD_REQUIRED ON
+		CXX_EXTENSIONS OFF )
+	target_compile_features( VulkanHppModule PUBLIC cxx_std_20 )
+	target_sources( VulkanHppModule
+		PUBLIC
+		FILE_SET vulkan_module_file BASE_DIRS ${CMAKE_SOURCE_DIR} TYPE CXX_MODULES FILES vulkan/vulkan.cppm )
+	target_include_directories( VulkanHppModule PUBLIC ${CMAKE_SOURCE_DIR} )
+	target_include_directories( VulkanHppModule PUBLIC "${CMAKE_SOURCE_DIR}/Vulkan-Headers/include" )
+endif()
 
 # The generator project !
 project( VulkanHppGenerator LANGUAGES CXX )
@@ -404,19 +418,4 @@ if( ${VULKAN_HPP_INSTALL} )
 	)
 	install( FILES ${VK_GENERATED_VULKAN_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/vulkan )
 
-endif()
-
-# Build Vulkan-Hpp as a module
-if( VULKAN_HPP_ENABLE_EXPERIMENTAL_CPP20_MODULES )
-	# create a target to provide VulkanHpp as C++20 module
-	add_library( VulkanHppModule )
-	set_target_properties( VulkanHppModule PROPERTIES
-		CXX_STANDARD_REQUIRED ON
-		CXX_EXTENSIONS OFF )
-	target_compile_features( VulkanHppModule PUBLIC cxx_std_20 )
-	target_sources( VulkanHppModule
-		PUBLIC
-		FILE_SET vulkan_module_file BASE_DIRS ${CMAKE_SOURCE_DIR} TYPE CXX_MODULES FILES vulkan/vulkan.cppm )
-	target_include_directories( VulkanHppModule PUBLIC ${CMAKE_SOURCE_DIR} )
-	target_include_directories( VulkanHppModule PUBLIC "${CMAKE_SOURCE_DIR}/Vulkan-Headers/include" )
 endif()

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -4742,9 +4742,9 @@ std::string VulkanHppGenerator::generateConstexprDefines() const
     return ${implementation};
   }
 )" };
-  auto const constexprCallTemplate     = std::string{ R"(  ${deprecated}VULKAN_HPP_CONSTEXPR auto ${constName} = ${callee}( ${arguments} );
+  auto const constexprCallTemplate     = std::string{ R"(  ${deprecated}VULKAN_HPP_CONSTEXPR_INLINE auto ${constName} = ${callee}( ${arguments} );
 )" };
-  auto const constexprValueTemplate    = std::string{ R"(  ${deprecated}VULKAN_HPP_CONSTEXPR ${type} ${constName} = ${value};
+  auto const constexprValueTemplate    = std::string{ R"(  ${deprecated}VULKAN_HPP_CONSTEXPR_INLINE ${type} ${constName} = ${value};
 )" };
   auto const deprecatedAttribute       = std::string{ R"(VULKAN_HPP_DEPRECATED("${reason}") )" };
 

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -718,7 +718,7 @@ export namespace VULKAN_HPP_NAMESPACE
 {
   ${usings}
 
-  export namespace VULKAN_HPP_RAII_NAMESPACE {
+  namespace VULKAN_HPP_RAII_NAMESPACE {
     ${raiiUsings}
   } // namespace VULKAN_HPP_RAII_NAMESPACE
 } // namespace VULKAN_HPP_NAMESPACE
@@ -5313,7 +5313,7 @@ std::string VulkanHppGenerator::generateCppModuleUsings() const
   auto const [enterDisableEnhanced, leaveDisableEnhanced] = generateProtection( "VULKAN_HPP_DISABLE_ENHANCED_MODE", false );
   usings += "\n" + enterDisableEnhanced + replaceWithMap( usingTemplate, { { "className", "StructExtends" } } ) + leaveDisableEnhanced + "\n";
 
-  auto const [enterDynamicLoader, leaveDynamicLoader] = generateProtection( "VULKAN_HPP_DYNAMIC_LOADER_TOOL" );
+  auto const [enterDynamicLoader, leaveDynamicLoader] = generateProtection( "VULKAN_HPP_ENABLE_DYNAMIC_LOADER_TOOL" );
   usings += "\n" + enterDynamicLoader + replaceWithMap( usingTemplate, { { "className", "DynamicLoader" } } ) + leaveDynamicLoader + "\n";
 
   usings += replaceWithMap( usingTemplate, { { "className", "DispatchLoaderDynamic" } } ) + "\n";

--- a/snippets/defines.hpp
+++ b/snippets/defines.hpp
@@ -86,6 +86,14 @@ constexpr int False = 0;
 #  define VULKAN_HPP_CONST_OR_CONSTEXPR const
 #endif
 
+#if !defined( VULKAN_HPP_CONSTEXPR_INLINE )
+#  if 201606L <= __cpp_inline_variables
+#    define VULKAN_HPP_CONSTEXPR_INLINE VULKAN_HPP_CONSTEXPR inline
+#  else
+#    define VULKAN_HPP_CONSTEXPR_INLINE VULKAN_HPP_CONSTEXPR
+#  endif
+#endif
+
 #if !defined( VULKAN_HPP_NOEXCEPT )
 #  if defined( _MSC_VER ) && ( _MSC_VER <= 1800 )
 #    define VULKAN_HPP_NOEXCEPT

--- a/tests/Cpp20Modules/CMakeLists.txt
+++ b/tests/Cpp20Modules/CMakeLists.txt
@@ -2,6 +2,9 @@ cmake_minimum_required( VERSION 3.25 )
 
 vulkan_hpp__setup_test( NAME Cpp20Modules CXX_STANDARD 20 LIBRARIES VulkanHppModule )
 
-if( NOT ( VULKAN_HPP_SAMPLES_BUILD_ONLY_DYNAMIC AND VULKAN_HPP_SAMPLES_BUILD_WITH_LOCAL_VULKAN_HPP ) )
+if( NOT VULKAN_HPP_SAMPLES_BUILD_ONLY_DYNAMIC )
 	target_link_libraries( Cpp20Modules PRIVATE Vulkan::Vulkan )
+	set_target_properties( Cpp20Modules PROPERTIES 
+		CXX_EXTENSIONS OFF
+	)
 endif()

--- a/vulkan/vulkan.cppm
+++ b/vulkan/vulkan.cppm
@@ -2764,9 +2764,9 @@ export namespace VULKAN_HPP_NAMESPACE
   using VULKAN_HPP_NAMESPACE::StructExtends;
 #endif /*VULKAN_HPP_DISABLE_ENHANCED_MODE*/
 
-#if defined( VULKAN_HPP_DYNAMIC_LOADER_TOOL )
+#if defined( VULKAN_HPP_ENABLE_DYNAMIC_LOADER_TOOL )
   using VULKAN_HPP_NAMESPACE::DynamicLoader;
-#endif /*VULKAN_HPP_DYNAMIC_LOADER_TOOL*/
+#endif /*VULKAN_HPP_ENABLE_DYNAMIC_LOADER_TOOL*/
 
   using VULKAN_HPP_NAMESPACE::DispatchLoaderDynamic;
 
@@ -2809,7 +2809,7 @@ export namespace VULKAN_HPP_NAMESPACE
   using VULKAN_HPP_NAMESPACE::isObsoletedExtension;
   using VULKAN_HPP_NAMESPACE::isPromotedExtension;
 
-  export namespace VULKAN_HPP_RAII_NAMESPACE
+  namespace VULKAN_HPP_RAII_NAMESPACE
   {
     //======================
     //=== RAII HARDCODED ===
@@ -2912,7 +2912,7 @@ export namespace VULKAN_HPP_NAMESPACE
     using VULKAN_HPP_RAII_NAMESPACE::BufferCollectionFUCHSIA;
 #endif /*VK_USE_PLATFORM_FUCHSIA*/
 
-    //=== VK_EXT_opacity_micromap ===
+       //=== VK_EXT_opacity_micromap ===
     using VULKAN_HPP_RAII_NAMESPACE::MicromapEXT;
 
     //=== VK_NV_optical_flow ===

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -212,6 +212,14 @@ constexpr int False = 0;
 #  define VULKAN_HPP_CONST_OR_CONSTEXPR const
 #endif
 
+#if !defined( VULKAN_HPP_CONSTEXPR_INLINE )
+#  if 201606L <= __cpp_inline_variables
+#    define VULKAN_HPP_CONSTEXPR_INLINE VULKAN_HPP_CONSTEXPR inline
+#  else
+#    define VULKAN_HPP_CONSTEXPR_INLINE VULKAN_HPP_CONSTEXPR
+#  endif
+#endif
+
 #if !defined( VULKAN_HPP_NOEXCEPT )
 #  if defined( _MSC_VER ) && ( _MSC_VER <= 1800 )
 #    define VULKAN_HPP_NOEXCEPT
@@ -6930,32 +6938,32 @@ namespace VULKAN_HPP_NAMESPACE
   //=========================================
   //=== CONSTEXPR CONSTANTs AND FUNCTIONs ===
   //=========================================
-  VULKAN_HPP_CONSTEXPR uint32_t AttachmentUnused                 = VK_ATTACHMENT_UNUSED;
-  VULKAN_HPP_CONSTEXPR uint32_t False                            = VK_FALSE;
-  VULKAN_HPP_CONSTEXPR float    LodClampNone                     = VK_LOD_CLAMP_NONE;
-  VULKAN_HPP_CONSTEXPR uint32_t LuidSize                         = VK_LUID_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxDescriptionSize               = VK_MAX_DESCRIPTION_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxDeviceGroupSize               = VK_MAX_DEVICE_GROUP_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxDriverInfoSize                = VK_MAX_DRIVER_INFO_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxDriverNameSize                = VK_MAX_DRIVER_NAME_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxExtensionNameSize             = VK_MAX_EXTENSION_NAME_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxGlobalPrioritySizeKhr         = VK_MAX_GLOBAL_PRIORITY_SIZE_KHR;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxMemoryHeaps                   = VK_MAX_MEMORY_HEAPS;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxMemoryTypes                   = VK_MAX_MEMORY_TYPES;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxPhysicalDeviceNameSize        = VK_MAX_PHYSICAL_DEVICE_NAME_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t MaxShaderModuleIdentifierSizeExt = VK_MAX_SHADER_MODULE_IDENTIFIER_SIZE_EXT;
-  VULKAN_HPP_CONSTEXPR uint32_t QueueFamilyExternal              = VK_QUEUE_FAMILY_EXTERNAL;
-  VULKAN_HPP_CONSTEXPR uint32_t QueueFamilyForeignExt            = VK_QUEUE_FAMILY_FOREIGN_EXT;
-  VULKAN_HPP_CONSTEXPR uint32_t QueueFamilyIgnored               = VK_QUEUE_FAMILY_IGNORED;
-  VULKAN_HPP_CONSTEXPR uint32_t Remaining3DSlicesExt             = VK_REMAINING_3D_SLICES_EXT;
-  VULKAN_HPP_CONSTEXPR uint32_t RemainingArrayLayers             = VK_REMAINING_ARRAY_LAYERS;
-  VULKAN_HPP_CONSTEXPR uint32_t RemainingMipLevels               = VK_REMAINING_MIP_LEVELS;
-  VULKAN_HPP_CONSTEXPR uint32_t ShaderUnusedKhr                  = VK_SHADER_UNUSED_KHR;
-  VULKAN_HPP_CONSTEXPR uint32_t SubpassExternal                  = VK_SUBPASS_EXTERNAL;
-  VULKAN_HPP_CONSTEXPR uint32_t True                             = VK_TRUE;
-  VULKAN_HPP_CONSTEXPR uint32_t UuidSize                         = VK_UUID_SIZE;
-  VULKAN_HPP_CONSTEXPR uint64_t WholeSize                        = VK_WHOLE_SIZE;
-  VULKAN_HPP_CONSTEXPR uint32_t HeaderVersion                    = VK_HEADER_VERSION;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t AttachmentUnused                 = VK_ATTACHMENT_UNUSED;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t False                            = VK_FALSE;
+  VULKAN_HPP_CONSTEXPR_INLINE float    LodClampNone                     = VK_LOD_CLAMP_NONE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t LuidSize                         = VK_LUID_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxDescriptionSize               = VK_MAX_DESCRIPTION_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxDeviceGroupSize               = VK_MAX_DEVICE_GROUP_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxDriverInfoSize                = VK_MAX_DRIVER_INFO_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxDriverNameSize                = VK_MAX_DRIVER_NAME_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxExtensionNameSize             = VK_MAX_EXTENSION_NAME_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxGlobalPrioritySizeKhr         = VK_MAX_GLOBAL_PRIORITY_SIZE_KHR;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxMemoryHeaps                   = VK_MAX_MEMORY_HEAPS;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxMemoryTypes                   = VK_MAX_MEMORY_TYPES;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxPhysicalDeviceNameSize        = VK_MAX_PHYSICAL_DEVICE_NAME_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t MaxShaderModuleIdentifierSizeExt = VK_MAX_SHADER_MODULE_IDENTIFIER_SIZE_EXT;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t QueueFamilyExternal              = VK_QUEUE_FAMILY_EXTERNAL;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t QueueFamilyForeignExt            = VK_QUEUE_FAMILY_FOREIGN_EXT;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t QueueFamilyIgnored               = VK_QUEUE_FAMILY_IGNORED;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t Remaining3DSlicesExt             = VK_REMAINING_3D_SLICES_EXT;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t RemainingArrayLayers             = VK_REMAINING_ARRAY_LAYERS;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t RemainingMipLevels               = VK_REMAINING_MIP_LEVELS;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t ShaderUnusedKhr                  = VK_SHADER_UNUSED_KHR;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t SubpassExternal                  = VK_SUBPASS_EXTERNAL;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t True                             = VK_TRUE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t UuidSize                         = VK_UUID_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint64_t WholeSize                        = VK_WHOLE_SIZE;
+  VULKAN_HPP_CONSTEXPR_INLINE uint32_t HeaderVersion                    = VK_HEADER_VERSION;
 
   template <typename T, typename = typename std::enable_if<std::is_integral<T>::value>::type>
   VULKAN_HPP_CONSTEXPR uint32_t apiVersionMajor( T const version )
@@ -7015,12 +7023,12 @@ namespace VULKAN_HPP_NAMESPACE
     return ( (uint32_t)(version)&0xFFFU );
   }
 
-  VULKAN_HPP_CONSTEXPR auto ApiVersion            = makeApiVersion( 0, 1, 0, 0 );
-  VULKAN_HPP_CONSTEXPR auto ApiVersion10          = makeApiVersion( 0, 1, 0, 0 );
-  VULKAN_HPP_CONSTEXPR auto ApiVersion11          = makeApiVersion( 0, 1, 1, 0 );
-  VULKAN_HPP_CONSTEXPR auto ApiVersion12          = makeApiVersion( 0, 1, 2, 0 );
-  VULKAN_HPP_CONSTEXPR auto ApiVersion13          = makeApiVersion( 0, 1, 3, 0 );
-  VULKAN_HPP_CONSTEXPR auto HeaderVersionComplete = makeApiVersion( 0, 1, 3, VK_HEADER_VERSION );
+  VULKAN_HPP_CONSTEXPR_INLINE auto ApiVersion            = makeApiVersion( 0, 1, 0, 0 );
+  VULKAN_HPP_CONSTEXPR_INLINE auto ApiVersion10          = makeApiVersion( 0, 1, 0, 0 );
+  VULKAN_HPP_CONSTEXPR_INLINE auto ApiVersion11          = makeApiVersion( 0, 1, 1, 0 );
+  VULKAN_HPP_CONSTEXPR_INLINE auto ApiVersion12          = makeApiVersion( 0, 1, 2, 0 );
+  VULKAN_HPP_CONSTEXPR_INLINE auto ApiVersion13          = makeApiVersion( 0, 1, 3, 0 );
+  VULKAN_HPP_CONSTEXPR_INLINE auto HeaderVersionComplete = makeApiVersion( 0, 1, 3, VK_HEADER_VERSION );
 
 }  // namespace VULKAN_HPP_NAMESPACE
 

